packages/pwa/app/commerce-api/hooks/useCustomerProductList.js

```diff
@@ -0,0 +1,325 @@
+import {useCommerceAPI, CustomerProductListsContext} from '../contexts'
+import {isError} from '../utils'
+
+import useCustomer from './useCustomer'
+
+/**
+ * This hook is designed to add customer product list capabilities
+ * to your app, it leverages the Commerce API - Shopper Customer endpoints.
+ * It uses a React context to store customer product list globally.
+ *
+ * By default, the Shopper Customer Product List API allows a shopper to
+ * save multiple product lists. However, the PWA only use a single
+ * product list, which is the wishlist. There is another hook useCustomerProductList,
+ * which is built on top of this hook to add wishlist specific logic.
+ * You can think this hook as the "parent class" for the useCustomerProductList hook.
+ *
+ * If your application need to handle multiple customer product lists, this
+ * is the hook for you, otherwise it is recommended to use useCustomerProductList hook.
+ */
+export function useCustomerProductLists() {
+    const api = useCommerceAPI()
+    const customer = useCustomer()
+    const {state, actions} = useContext(CustomerProductListsContext)
+
+    const self = useMemo(() => {
+        return {
+            isInitialized: state.isInitialized,
+            data: state.productLists,
+
+            reset() {
+                actions.reset()
+            },
+
+            async getLists() {
+                const response = await api.shopperCustomers.getCustomerProductLists({
+                    parameters: {
+                        customerId: customer.customerId
+                    }
+                })
+
+                if (isError(response)) {
+                    throw new Error(response)
+                }
+
+                return response
+            },
+
+            async createList(name, options) {
+                const response = await api.shopperCustomers.createCustomerProductList({
+                    body: {
+                        ...options,
+                        name
+                    },
+                    parameters: {
+                        customerId: customer.customerId
+                    }
+                })
+
+                if (isError(response)) {
+                    throw new Error(response)
+                }
+
+                return response
+            }
+        }
+    }, [customer.customerId, state])
+    return self
+}
+
+/**
+ * This hook is the "child class" of useCustomerProductLists.
+ * It provides functionalities to manage a single wishlist for shoppers.
+ */
+export function useCustomerProductList(name, options = {}) {
+    const api = useCommerceAPI()
+    const customer = useCustomer()
+    const _super = useCustomerProductLists()
+    const {actions} = useContext(CustomerProductListsContext)
+    const DEFAULT_LIST_TYPE = 'wish_list'
+    const type = options?.type || DEFAULT_LIST_TYPE
+
+    const self = useMemo(() => {
+        return {
+            isInitialized: _super.isInitialized,
```

Comment:
This surely can't be true. This will return `true` then the `CustomerProductLists` are loaded, which may or may not include the list specified by `name`.  Should you be checking it that list exists?